content/keps.md

```diff
@@ -0,0 +1,148 @@
+---
+title: "Collaboration with Kubernetes SIG-Node"
+---
+
+containerd is a popular container runtime used with
+[Kubernetes](https://kubernetes.io).  containerd implements the Kubernetes
+Container Runtime Interface (CRI) and is expected to participate in the
+development and implementation of CRI.
+
+Changes to Kubernetes (including CRI) are primarily driven through the
+Kubernetes Enhancement Proposal (KEP) process.  KEPs are proposed to individual
+Special Interest Groups (SIGs) within the Kubernetes project, which help to
+refine, clarify, and decide the content of an individual KEP as well as drive
+its implementation.  CRI-related KEPs are primarily handled through SIG-Node.
+SIG-Node has
+[adopted a policy](https://www.kubernetes.dev/docs/code/cri-api-dev-policies/)
+which requires KEPs to reach maturity in both containerd and CRI-O before
+advancing through graduation stages.
+
+This document describes the process that the containerd project uses to
+collaborate with SIG-Node on any KEP or other CRI-related change.
+
+## Goals
+
+* Provide visibility of SIG-Node integration work for containerd (including
+  KEPs) for:
+  * containerd maintainers and community members who are not directly involved
+    in SIG-Node
+  * SIG-Node participants who are not involved in the containerd community
+* Ensure containerd maintainers and communtiy members have adequate
+  opportunities to participate in design without requiring full context on
+  every SIG-Node KEP or who are not familiar with the KEP process
+* Ensure contributors understand the status of their proposed contributions
+* Spread the KEP tracking and review load among interested maintainers, and
+  enable at least two interested containerd maintainers per KEP or
+  Kubernetes-related work item in order to facilitate knowledge sharing
+
+## Non-goals
+
+* Change the development or release process of containerd.  See the
+  [contribution guidelines]({{< ref "contributing.md" >}}) and
+  [versioning and release]({{< ref "releases.md" >}}) pages.
+* Guarantee KEP delivery, outside of providing feedback during KEP design,
+  guiding contributors through processes, and reviews.
+
+## Roles
+
+* SIG-Node member liaison - a containerd maintainer responsible for
+  facilitating communication between SIG-Node and the containerd project
+  regarding KEP status, targeted containerd releases, and any other integration
+  work or pain points.
+* KEP shepherd - for each KEP, at least one (preferably two) containerd
+  maintainer(s) responsible for helping a KEP move through the implementation
+  and release process in containerd.
+
+## Metadata
+
+* Per-KEP issue in the
+  [containerd/containerd](https://github.com/containerd/containerd) repository
+  for tracking KEPs that require CRI changes in the container
+  runtime or one of our other containerd subprojects.  (This includes
+  relatively minor changes, such as a runc update or checking containerd config
```

Comment:
"runc update" means updating the `/usr/local/sbin/runc` binary (for the CI) or the `runc update` command?

At least the former one rarely relates to a KEP especially for patch releases, and does not seem to need an issue ticket.

Reply:
@mikebrow suggested this set of examples; I think it means updating the runc binary.

Reply:
yes I meant even the minor issue of updating the runc-version file over in script/ in this case... 

I have agreed with the point that it does not seem we need an issue ticket in every case.. Believe we had a thread where you and I agreed some changes are so easy and obvious that we just get it done in a quick PR and you can mark that as done over in the actual KEP issue in k/enhancements where it's being discussed. 

After a long discussion with Sam I was swayed to agreement for the process where even for the easy stuff, if there is something to track in the KEP description/PR then we should open an issue in c/c. "just in case" For example someone may know of an issue with moving up to that version of runc that would delay merge.